src/Promitor.Integrations.LogAnalytics/LogAnalyticsClient.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Monitor.Query;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Extensions.Logging;
+
+namespace Promitor.Integrations.LogAnalytics
+{
+    public class LogAnalyticsClient
+    {
+        private readonly ILogger _logger;
+        public readonly string ColumnNameResult = "result";
+        private readonly Uri _defaultEndpoint = new Uri("https://api.loganalytics.io");
+        private readonly Uri _govEndpoint = new Uri("https://api.loganalytics.us");
+
+        private readonly LogsQueryClient _logsQueryClient;
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            _logsQueryClient = new LogsQueryClient(new DefaultAzureCredential());
+        }
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory, AzureEnvironment azureEnvironment, TokenCredential tokenCredentials)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            var uri = _defaultEndpoint;
+            if (azureEnvironment.Name == nameof(AzureEnvironment.AzureUSGovernment))
+                uri = _govEndpoint;
+
+            _logsQueryClient = new LogsQueryClient(uri, tokenCredentials);
+        }
+
+        public async Task<double> QueryDouble(string workspaceId, string query, TimeSpan aggregationInterval)
+        {
+            var queryResult = await _logsQueryClient.QueryWorkspaceAsync(workspaceId, query, new QueryTimeRange(aggregationInterval, DateTime.UtcNow));
+            if (queryResult.Value.Table.Rows.Count != 1)
+            {
+                _logger.LogError("Query result length need to be 1");
+                throw new Exception("Query result length need to be 1");
+            }
+
+            var result = queryResult.Value.Table.Rows[0].GetDouble(ColumnNameResult);
```

Comment:
We are doing this without checking if there even is a row

Reply:
We already check the query result needs to have only 1 row above

Reply:
Good poitn